### ISSUE TITLE
Error with Numpy 2.0.0rc1

### ISSUE THREAD

**Issue by astrofrog** [astropy/ccdproc#818]
I'm running into the following error with Numpy 2.0.0rc1:

```
    from .extern.bitfield import bitfield_to_boolean_mask as _bitfield_to_boolean_mask
.tox/py311-ccdproc/lib/python3.11/site-packages/ccdproc/extern/bitfield.py:56: in <module>
    MAX_UINT_TYPE = np.maximum_sctype(np.uint)
.tox/py311-ccdproc/lib/python3.11/site-packages/numpy/__init__.py:397: in __getattr__
    raise AttributeError(
E   AttributeError: `np.maximum_sctype` was removed in the NumPy 2.0 release. Use a specific dtype instead. You should avoid relying on any implicit mechanism and select the largest dtype of a kind explicitly in the code.
```

**Comment by mwcraig** [comment]
Thanks for reporting this -- I'll get it fixed next week.

**Comment by neutrinoceros** [comment]
@mwcraig we need a new release before numpy 2 comes out and this starts affecting users. Can you give us an estimation of when it could happen ? Thanks !

**Comment by mwcraig** [comment]
@neutrinoceros -- I'll do my best to get it out today.....

**Comment by mwcraig** [comment]
@neutrinoceros -- apologies that "today" lasted two weeks 😬...I just released 2.4.2 on pypi

**Comment by neutrinoceros** [comment]
No worries at all, and thanks for the ping !
